forked-daapd.conf.in

```diff
@@ -53,6 +53,9 @@ general {
 	# unusual platform and experience audio drop-outs, you can try changing
 	# this option
 #	high_resolution_clock = yes
+
+	# Set the MP3 streaming bit rate (in kbps), valid options: 64 / 96 / 128 / 192 / 320
```

Comment:
I think the config option should have its own section. Then we can later add other options. Or you could add say sample_rate right away. Also a better description is needed, otherwise users that don't use /stream.mp3 will be confused by the meaning of this option.

@chme is this change perhaps a candidate for using the new settings feature? If you think so, could you then give some advice on how to do it?

Reply:
@ejurgensen - to allow the users to specify a bit_rate (or perhaps even other params) the server could receive `/stream.mp3?bit_rate=192`.  This would take president over the server config value that we'll hold in `streaming_quality_out.bit_rate` and use this for setting up the mp3 encode for this client.

The web UI would let the user set the required `bit_rate` on the settings page that is then sent to the server.

Does that work for you?

Reply:
Just had another dive into this and it looks like I misunderstood the setup for the transcode and that **theres only one** `transcode_encode_ctx` no matter how many streaming clients we have.

Given this, does it make sense to allow (multiple) streaming clients to potentially request a different set of transcode parameters etc?

Reply:
I've got this `/stream.mp3?bit_rate=xxx` working in another branch with this commit 4c69f5435897d6a6bc84d3eb8f047d198d7f0ca8.  You can start playback and attach a client:
```
ffmpeg -i http://forked-daapd.local:3689/stream.mp3 -f pulse "foo"
```
and you see'll the strream come in with the server defaulted bit rate. In another terminal when you request:
```
ffmpeg -i http://forked-daapd.local:3689/stream.mp3?bit_rate=32 -f pulse "foo"
```
you'll hear that both streams are now at 32kbps (ie awkful so you hear the drop in quality).  With this in place another PR can add the UI settings change (which does not need to be saved in db like the `composer`).

Let me know if you want this work to be part of this PR

Reply:
I didn't really ask for the above in my review, and I don't agree with it. Like you say a bit_rate parameter entails multiple encodings, and I don't think we should do that. Also I don't think icymetaint should be configurable (it is too low level), and I don't suppose setting bits_per_sample has any meaning when it comes to mp3? And what about the channel param - can ffmpeg actually encode mp3 to more channels than 2?

BTW I have received something like 8 email notifications on this PR alone since my review. That is a bit much. It would be nice if you for instance pushed all changes instead of splitting them.

Reply:
So we drop `bits_per_sample` and `channels` from being configurable and leave them to 16/2 as they are now.

I’m a little hesitant from removing `icy_metaint` as I found that my Roku Soundbridge didn’t like values above 16k - no meta displayed - so I wonder if there are other clients that have such restrictions out there too that may need tweaking this value.  Could we leave this as configurable but remove it from the ‘forked-daapd.conf.in` so it becomes a _hidden_ option like some others I saw in ‘`conffile.c`?

Reply:
Yes that is ok